src/vm/comutilnative.cpp

```diff
@@ -899,15 +899,15 @@ UINT64   GCInterface::m_remPressure[NEW_PRESSURE_COUNT] = {0, 0, 0, 0};   // his
 // (m_iteration % NEW_PRESSURE_COUNT) is used as an index into m_addPressure and m_remPressure
 UINT     GCInterface::m_iteration = 0;
 
-FCIMPL5(void, GCInterface::GetMemoryInfo, UINT32* highMemLoadThreshold, UINT64* totalPhysicalMem, UINT32* lastRecordedMemLoad, size_t* lastRecordedHeapSize, size_t* lastRecordedFragmentation)
+FCIMPL1(void, GCInterface::GetMemoryInfo, GCMemoryInfo* memoryInfo)
 {
     FCALL_CONTRACT;
 
     FC_GC_POLL_NOT_NEEDED();
     
-    return GCHeapUtilities::GetGCHeap()->GetMemoryInfo(highMemLoadThreshold, totalPhysicalMem, 
-                                                       lastRecordedMemLoad, 
-                                                       lastRecordedHeapSize, lastRecordedFragmentation);
+    return GCHeapUtilities::GetGCHeap()->GetMemoryInfo((uint32_t*)&memoryInfo->m_highMemoryLoadThreshold, (uint64_t*)&memoryInfo->m_totalAvailableMemory,
+                                                       (uint32_t*)&memoryInfo->m_memoryLoad,
+                                                       (size_t*)&memoryInfo->m_heapSize, (size_t*)&memoryInfo->m_fragmentation);
```

Comment:
The upper half of `m_heapSize` is going to be left uninitialized on 32-bit platforms.

Reply:
I have seen many bugs (uninitialized data, buffer overrun or endianess problem) caused by pattern like this. I think it is better to do an extra copy and cast the value, like:

```
size_t heapSize;
...
GCHeapUtilities::GetGCHeap()->GetMemoryInfo(... , &heapSize, ...);
...
memoryInfo->m_heapSize = heapSize;
```